typescript/web/src/components/dataset-classes/upsert-class-modal/upsert-class-modal.tsx

```diff
@@ -112,6 +117,8 @@ const useModalState = ({ isOpen, onClose }: UpsertClassModalProps) => {
   const { editClass, datasetId, datasetSlug } = useDatasetClasses();
   const [classNameInputValue, setClassNameInputValue] = useState<string>("");
   const [errorMessage, setErrorMessage] = useState<string>("");
+  const [isClassCreationPending, setIsClassCreationPending] =
+    useState<boolean>(false);
```

Comment:
Using a dedicated state would trigger additional re-renders which are avoidable by relying on the `loading` variable which is returned by `useMutation`.
If the `createLabelClass` and `updateLabelClass` (whose loading state should have been handled too) don't use `useQuery` yet, just change the code to use this hook or find an alternative based on `client.mutate`.

Reply:
Refactoring this code to properly use `useMutation` instead of `ApolloClient.mutate` would require quite a lot of work in its current state, I suggest we:
- merge this pull request as a quick fix for now.
- create a new issue to ask for a proper refactor of the class upsert code from the class list page.

Reply:
It's not that long actually, I've done in in ~30mn. See the last commit I've just pushed 😉

Reply:
I see, if I did it myself, being not too familiar with react hooks yet, I would have spent quite some time checking that I didn't do anything wrong though :P